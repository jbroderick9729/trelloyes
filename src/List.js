import React, { Component } from 'react';
import Card from './Card.js';
import './List.css';

export default function List(props) {
        return (
            <div>
                <section className="List">
                    <header className="List-header">
                        <h2>{props.header}</h2>
                    </header>
                        <div className="List-cards">
                            {props.cards.map((item) => (
                                <Card key={item.id} title={item.title} content={item.content} />
                                ))}
                        </div>
                </section>
            </div>
        )
}