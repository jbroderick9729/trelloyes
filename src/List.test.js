import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe('List component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List key={1} header="Test Header" cards={ [{'a': { title: 'Test card', content: 'test content' }}]} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
        const tree = renderer.create(<List key={1} header="Test Header" cards={ [{'a': { title: 'Test card', content: 'test content' }}]} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});