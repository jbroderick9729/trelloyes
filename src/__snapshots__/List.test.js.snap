// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<div>
  <section
    className="List"
  >
    <header
      className="List-header"
    >
      <h2>
        Test Header
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div
        className="Card"
      >
        <h3 />
        <p />
        <button>
          Delete
        </button>
      </div>
    </div>
  </section>
</div>
`;
